XPlat/Shopify/iOS/source/Shopify.iOS/ApiDefinition.cs

```diff
@@ -0,0 +1,1964 @@
+
+		[Static]
+		[Export ("completeCheckoutFromLaunchURL:")]
+		void CompleteCheckout (NSUrl url);
+
+	}
+
+	[BaseType (typeof(ViewController), Name = "BUYProductViewController")]
+	interface ProductViewController : Themeable
+	{
+		[Export ("initWithClient:theme:")]
+		IntPtr Constructor (BuyClient client, Theme theme);
+
+		[Export ("loadProduct:completion:"), Async]
+		void LoadProduct (string productId, Action<bool, NSError> completion);
+
+		[Export ("loadWithProduct:completion:"), Async]
+		void LoadProduct (Product product, Action<bool, NSError> completion);
+
+		[Export ("productId", ArgumentSemantic.Strong)]
+		string ProductId { get; }
+
+		[Export ("product", ArgumentSemantic.Strong)]
+		Product Product { get; }
+
+		[Export ("isLoading")]
+		bool IsLoading { get; }
+
+		[Export ("presentPortraitInViewController:")]
+		void PresentPortraitInViewController (UIViewController controller);
+
+	}
+
+	[BaseType (typeof(UIView), Name = "BUYProductViewErrorView")]
+	interface ProductViewErrorView
+	{
+		[Export ("errorLabel", ArgumentSemantic.Strong)]
+		UILabel ErrorLabel { get; set; }
+
+		[Export ("hiddenConstraint", ArgumentSemantic.Strong)]
+		NSLayoutConstraint HiddenConstraint { get; set; }
+
+		[Export ("visibleConstraint", ArgumentSemantic.Strong)]
+		NSLayoutConstraint VisibleConstraint { get; set; }
+
+		[Export ("initWithTheme:")]
+		IntPtr Constructor (Theme theme);
+
+	}
+
+	[BaseType (typeof(UIView), Name = "BUYProductViewFooter")]
+	interface ProductViewFooter
+	{
+		[Export ("initWithTheme:showApplePaySetup:")]
+		IntPtr Constructor (Theme theme, bool showApplePaySetup);
+
+		[Export ("checkoutButton", ArgumentSemantic.Strong)]
+		CheckoutButton CheckoutButton { get; set; }
+
+		[Export ("buyPaymentButton", ArgumentSemantic.Strong)]
+		PaymentButton BuyPaymentButton { get; set; }
+
+		[Export ("setApplePayButtonVisible:")]
+		void SetApplePayButtonVisible (bool isApplePayAvailable);
+
+		[Export ("updateButtonsForProductVariant:")]
+		void UpdateButtonsForProductVariant (ProductVariant productVariant);
+
+	}
+
+	[BaseType (typeof(UIView), Name = "BUYProductViewHeader")]
+	interface ProductViewHeader
+	{
+		[Export ("collectionView", ArgumentSemantic.Strong)]
+		UICollectionView CollectionView { get; set; }
+
+		[Export ("productViewHeaderOverlay", ArgumentSemantic.Strong)]
+		ProductViewHeaderOverlay ProductViewHeaderOverlay { get; set; }
+
+		[Export ("initWithFrame:theme:")]
+		IntPtr Constructor (CGRect frame, Theme theme);
+
+		[Export ("imageHeightWithScrollViewDidScroll:")]
+		nfloat ImageHeightWithScrollViewDidScroll (UIScrollView scrollView);
+
+		[Export ("setCurrentPage:")]
+		void SetCurrentPage (nint currentPage);
+
+		[Export ("setImageForSelectedVariant:withImages:")]
+		void SetImageForSelectedVariant (ProductVariant productVariant, Image[] images);
+
+	}
+
+	[BaseType (typeof(UIView), Name = "BUYProductViewHeaderBackgroundImageView")]
+	interface ProductViewHeaderBackgroundImageView
```

Comment:
Checking with header, I would suggest to add [DisableDefaultCtor] attribute because it would make it easier to configure the theme using Constructor (Theme theme).